core/src/main/java/io/seata/core/raft/RaftServerFactory.java

```diff
@@ -0,0 +1,133 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.core.raft;
+
+import java.io.IOException;
+import com.alipay.sofa.jraft.conf.Configuration;
+import com.alipay.sofa.jraft.entity.PeerId;
+import com.alipay.sofa.jraft.option.NodeOptions;
+import io.seata.common.loader.EnhancedServiceLoader;
+import io.seata.common.util.StringUtils;
+import io.seata.config.ConfigurationFactory;
+import io.seata.core.constants.ConfigurationKeys;
+import io.seata.core.store.StoreMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import static io.seata.common.DefaultValues.SEATA_RAFT_GROUP;
+import static io.seata.core.raft.AbstractRaftServer.RAFT_TAG;
+
+/**
+ * @author funkye
+ */
+public class RaftServerFactory {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RaftServerFactory.class);
+
+    private AbstractRaftServer raftServer;
+
+    private AbstractRaftStateMachine stateMachine;
+
+    private Boolean raftMode = false;
+
+    public static RaftServerFactory getInstance() {
+        return SingletonHandler.instance;
+    }
+
+    public void init(String host, int port, String... defaultConf) throws IOException {
+        String initConfStr = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.SERVER_RAFT_CLUSTER);
+        if (StringUtils.isBlank(initConfStr)) {
+            if (defaultConf == null || defaultConf.length == 0) {
+                if (LOGGER.isWarnEnabled()) {
+                    LOGGER.warn("initialize SofaJraft fail cluster is null");
+                }
+                return;
+            } else {
+                initConfStr = defaultConf[0];
+            }
+        }
+        String mode = ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.STORE_MODE);
+        StoreMode storeMode = StoreMode.get(mode);
+        if (storeMode.equals(StoreMode.RAFT)) {
+            raftMode = true;
+        }
+        String colon = ":";
+        int constantInt = 100 * 10;
+        String serverIdStr = host + colon + (port - constantInt);
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info("initialize SofaJraft");
+        }
+        final String dataPath =
+            ConfigurationFactory.getInstance().getConfig(ConfigurationKeys.STORE_FILE_DIR, "sessionStore") + "/"
+                + serverIdStr.split(colon)[1];
+        final NodeOptions nodeOptions = new NodeOptions();
+        // Set the election timeout to 1 second
+        nodeOptions.setElectionTimeoutMs(constantInt);
+        // Close the CLI service.
+        nodeOptions.setDisableCli(false);
+        // Snapshot should be made every 30 seconds
+        nodeOptions.setSnapshotIntervalSecs(60);
```

Comment:
30?

Reply:
change to open configuration items later

Reply:
change to open configuration items later